lib/utils/report.mjs

```diff
@@ -146,3 +134,80 @@ function printfLike(format, ...args) {
 		return result.replace(/%[ds]/, String(arg));
 	}, format);
 }
+
+/** @typedef {import('stylelint').DisabledRangeObject} DisabledRangeObject */
+/** @typedef {import('stylelint').StylelintPostcssResult} StylelintPostcssResult */
+/** @typedef {import('stylelint').Range} Range */
+/** @typedef {import('stylelint').Problem} Problem */
+
+/**
+ * @param {string} ruleName
+ * @param {number} startLine
+ * @param {DisabledRangeObject} disabledRanges
+ */
+function isDisabled(ruleName, startLine, disabledRanges) {
+	if (!disabledRanges) return false;
+
+	const ranges = disabledRanges[ruleName] ?? disabledRanges[RULE_NAME_ALL] ?? [];
+
+	for (const range of ranges) {
+		if (
+			// If the problem is within a disabledRange,
+			// and that disabledRange's rules include this one
+			range.start <= startLine &&
+			(range.end === undefined || range.end >= startLine) &&
+			(!range.rules || range.rules.includes(ruleName))
+		) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+/** @param {Problem} problem */
+function isFixApplied(problem) {
+	const { end, start, index, endIndex, fix, node, result, ruleName } = problem;
+	const hasRange = start && isNumber(end?.column);
+	const hasIndexes = isNumber(index) && isNumber(endIndex);
+	const hasRangeData = hasRange || hasIndexes;
+
+	if (typeof fix !== 'function') return false;
```

Comment:
wip
minor

`addFixdata` for unmigrated rules (report path) and migrated rules (unfixable path)